examples/python/trigger-pagerduty-incident/handler/handler.py

```diff
@@ -0,0 +1,143 @@
+            sys.exit(1)
+        except KeyError as e:
+            print(f'mandatory configuration key not found: {e}')
+            sys.exit(1)
+        self.session=conn
+
+    # PagerDuty REST API implementation        
+    def connect(self,obj):
+        """make a rest api call to Pagerduty Events API
+        
+        Arguments:
+
+            obj {dict} -- Generated API Body for the PagerDuty Events API
+
+        Returns:
+            FaaSResponse -- status code and message
+        """
+        obj['routing_key'] = self.routingkey
+        obj['event_action'] = self.event_action
+        try:
+            resp = self.session.post(PAGERDUTY_API_PATH,json=obj)
+            resp.raise_for_status()
+            print(resp.text)
+            resp_body = json.loads(resp.text)
+            return FaaSResponse('200', 'successfully {0}ed event action with dedup_key: {1}'.format(self.event_action, resp_body['dedup_key']))
+        except requests.HTTPError as err:
+            return FaaSResponse('500', 'could not invoke Pagerduty API {0}'.format(err))
+
+def handle(req):
+    # Validate input
+    try:
+        ref = json.loads(req)
+    except ValueError as err:
+        res = FaaSResponse('400','invalid JSON {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+    
+    # Assert that the Event conforms to CloudEvent Spec
+    # For debugging: validate the JSON blob we received - uncomment if needed
+    # print(j)
+    try:
+        eventdata = (ref['data'])
+    except KeyError as err:
+        res = FaaSResponse('400','JSON does not conform to CloudEvent spec {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+
+    # Map the CloudEvent data and build the PagerDuty Event API Request body
+    obj = {
+            'payload': {
+                'summary': eventdata['FullFormattedMessage'],
+                'timestamp': eventdata['CreatedTime'],
+                'source': ref['source'],
+                'severity': 'info',
+                'component': eventdata['Vm']['Name'],
+                'group': eventdata['Host']['Name'],
+                'class': ref['subject'],
+                'custom_details': {
+                    'user': eventdata['UserName'],
+                    'Datacenter': eventdata['Datacenter'],
+                    'ComputeResource': eventdata['ComputeResource'],
+                    'Host': eventdata['Host'],
+                    'VM': eventdata['Vm']
+                }
+            },
+            'routing_key': 'REPLACE',
+            'images': [{
+                'src': 'https://www.pagerduty.com/wp-content/uploads/2016/05/pagerduty-logo-green.png',
+                'href': 'https://example.com/',
+                'alt': 'Image of VM'
+            }],
+            'links': [{
+                'href': 'https://example.com/',
+                'text': 'Link to VM'
+            }],
+            'event_action': 'REPLACE',
+            'client': 'vCenter',
```

Comment:
To discuss: should the client be VEBA for better troubleshooting? vCenter source is already covered in the payload earlier.

Reply:
for the api body w/
```
  "client": "vCenter",
  "client_url": "https://10.0.10.90/sdk",
```

here is an example of how the client value gets rendered within PagerDuty
![image](https://user-images.githubusercontent.com/30903187/79053058-3d0ec600-7bef-11ea-8465-b929fd234130.png)

Agreed on the client being VEBA but the question is around the URL, what should we be using for the client URL?